File: HRM.backend/src/HRM.Core/Entities/EmployeeProfile/Dependent.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using HRM.backend.src.HRM.Core.Enums;

namespace HRM.backend.src.HRM.Core.Entities.EmployeeProfile
{
    [Table("dependents")]
    public class Dependent
    {
        [Key] public int Id { get; set; }

        public int? EmployeeId { get; set; }
        [ForeignKey("EmployeeId")]
        public virtual Employee? Employee { get; set; }

        [StringLength(100)] public required string FullName { get; set; }
        public DependentRelation Relationship { get; set; }

        [StringLength(20)] public string? IdNumber { get; set; }
        [StringLength(20)] public string? TaxDependentCode { get; set; } // Mã số thuế phụ thuộc

        public DateTime? BirthDate { get; set; }
        public DateTime ValidFrom { get; set; }
        public DateTime? ValidTo { get; set; }
        public bool IsActive { get; set; } = true;
        public string? EvidenceUrl { get; set; }
        public string? Note { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}

File: HRM.frontend/src/features/employees/types/dependent.ts
export type DependentRelation = 0 | 1 | 2 | 3;

export interface DependentDto {
  id: number;
  employeeId: number;
  fullName: string;
  relationship: DependentRelation;
  idNumber: string | null;
  taxDependentCode: string | null;
  birthDate: string | null;
  validFrom: string;
  validTo: string | null;
  isActive: boolean;
  evidenceUrl: string | null;
  note: string | null;
}

export interface PendingDependentRequest {
  id: number;
  employeeId: number;
  employeeName: string;
  employeeCode: string;
  dependentId: number | null;
  actionType: "CREATE" | "UPDATE" | "DEACTIVATE" | string;
  requestedDataJson: string;
  evidenceUrl: string | null;
  status: string;
  createdAt: string;
}

export interface DependentFormState {
  fullName: string;
  relationship: DependentRelation;
  idNumber: string;
  taxDependentCode: string;
  birthDate: string;
  validFrom: string;
  validTo: string;
  note: string;
  evidenceFile: File | null;
}
